app/models/map.rb

```diff
@@ -23,4 +24,12 @@ class Map < ApplicationRecord
 
   belongs_to :user
   has_many :locations, dependent: :destroy
+  has_many :payload_dumps, class_name: "Sync::PayloadDump", dependent: :destroy
+  has_one :api_key, dependent: :destroy
+
+  after_create :create_api_key
+
+  def create_api_key
+    ApiKey.create(map: self)
```

Comment:
Could add a little "return if exist, otherwise create new one" here. Soemthing like:

```
def create_api_key
   api_key || ApiKey.create(map: self)
end
```

This way we can call this method anywhere in the code, it will not try to create more than one api key and raise a validation error. In the rake task for example, we also then can get rid of the `unless map.api_key.present?`

Reply:
Added in 86d8dce